types/postProcess.go

```diff
@@ -16,7 +17,20 @@ func (crud *servicePostAction) Create(_ context.Context, args ...crud.Arg) (crud
 }
 
 func (crud *servicePostAction) Delete(_ context.Context, args ...crud.Arg) (crud.Arg, error) {
-	return nil, crud.currentState.Services.Delete(*((args[0].(*state.Service)).ID))
+	serviceID := *args[0].(*state.Service).ID
+
+	// Delete all plugins associated with this service as that's the implicit behavior of Kong (cascade delete).
+	plugins, err := crud.currentState.Plugins.GetAllByServiceID(serviceID)
+	if err != nil {
+		return nil, fmt.Errorf("error looking up plugins for service '%v': %v", serviceID, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error looking up plugins for service '%v': %w", serviceID, err)
```

Reply:
Hi !
Adding [errorlint](https://golangci-lint.run/usage/linters/#errorlint) linter to golangci-lint could help you ensure that .

Reply:
+1, will look to add this one

Reply:
ref: https://github.com/Kong/deck/pull/923